File: src/main/java/mods/orca/mffs/items/ItemFreqcard.java
package mods.orca.mffs.items;

import net.minecraft.client.util.ITooltipFlag;
import net.minecraft.item.ItemStack;
import net.minecraft.world.World;

import javax.annotation.Nullable;
import java.util.List;

public class ItemFreqcard extends ItemBase {
    public static final String KEY_CORE_POS = "corePos";

    public ItemFreqcard() {
        super("freqcard", false);
    }

    @Override
    public void addInformation(ItemStack stack, @Nullable World worldIn, List<String> tooltip, ITooltipFlag flagIn) {
        if (stack.hasTagCompound() && stack.getTagCompound().hasKey(KEY_CORE_POS)) {
            int[] corePos = stack.getTagCompound().getIntArray(KEY_CORE_POS);
            tooltip.add("Linked to core: " + corePos[0] + ", " +corePos[1] + ", " + corePos[2]);
        }
    }
}

File: src/main/java/mods/orca/mffs/items/ItemFreqcard.kt
package mods.orca.mffs.items

import net.minecraft.client.util.ITooltipFlag
import net.minecraft.item.ItemStack
import net.minecraft.world.World

class ItemFreqcard : ItemBase("freqcard", false) {

    companion object {
        const val KEY_CORE_POS: String = "corePos"
    }

    override fun addInformation(
        stack: ItemStack,
        worldIn: World?,
        tooltip: MutableList<String>,
        flagIn: ITooltipFlag
    ) {

        // TODO: Rewrite this system to track cores by ID rather than world pos, which may be considerably slow.

        stack.tagCompound
            ?.takeIf { it.hasKey(KEY_CORE_POS) }
            ?.run { getIntArray(KEY_CORE_POS) }
            ?.run { tooltip.add("Linked to core: ${get(0)}, ${get(1)}, ${get(2)}") }

    }

}
